org.civicrm.reqangular/test/models/relationship.model.spec.js

```diff
@@ -0,0 +1,105 @@
+/* eslint-env amd, jasmine */
+
+define([
+  'common/lodash',
+  'common/mocks/data/relationship.data',
+  'common/angularMocks',
+  'common/models/relationship.model',
+  'common/models/instances/relationship.instance',
+  'common/services/api/relationship.api'
+], function (_, relationshipData) {
+  'use strict';
+
+  describe('RelationshipModel', function () {
+    var $q, $rootScope, RelationshipModel, RelationshipAPI, RelationshipInstance,
+      result;
+    var filters = { key: 'filters' };
+    var pagination = { key: 'pagination' };
+
+    beforeEach(module('common.models', 'common.models.instances'));
+
+    beforeEach(inject(function (_$q_, _$rootScope_,
+      _RelationshipAPI_, _RelationshipInstance_, _RelationshipModel_) {
+      $q = _$q_;
+      $rootScope = _$rootScope_;
+      RelationshipAPI = _RelationshipAPI_;
+      RelationshipInstance = _RelationshipInstance_;
+      RelationshipModel = _RelationshipModel_;
+
+      spyOn(RelationshipAPI, 'all').and.returnValue($q.resolve({
+        list: relationshipData.all.values
+      }));
+    }));
+
+    it('has the expected api', function () {
+      expect(Object.keys(RelationshipModel)).toEqual(['all', 'allValid']);
+    });
+
+    describe('all()', function () {
+      beforeEach(function () {
+        RelationshipModel.all(filters, pagination)
```

Comment:
Please use https://compucorp.atlassian.net/wiki/spaces/PCHR/pages/121371880/AngularJS#AngularJS-Testingasynchronousmethods when testing async methods.
`done` is missing here.